index.html

```diff
@@ -2122,41 +2113,39 @@
 destruction.
 
 An [=actor=] <dfn data-lt="share|sharing">shares</dfn> data if it provides it to any other
-[=actor=]. Note that, under this definition, an [=actor=] that provides data to its own
+[=data controller=]. Note that, under this definition, an [=actor=] that provides data to its own
 [=service providers=] is not [=sharing=] it.
 
-An [=actor=] <dfn data-lt="sell|selling">sells</dfn> data when it [=shares=] it in exchange
-for consideration, monetary or otherwise.
+An [=actor=] <dfn data-lt="sell|selling">sells</dfn> data when it [=shares=] the data in exchange
+for something of value, even if that value isn't monetary.
 
 The <dfn>purpose</dfn> of a given [=processing=] of data is an anticipated, intended, or
 planned outcome of this [=processing=] which is achieved or aimed for within a given
-[=context=]. A [=purpose=], when described, should be specific enough to be actionable by
-someone familiar with the relevant [=context=] (ie. they could independently determine
-[=means=] that reasonably correspond to an implementation of the [=purpose=]).
+[=context=]. A [=purpose=], when described, should be specific enough that
+someone familiar with the relevant [=context=] could pick some
+[=means=] that would achieve the [=purpose=].
 
-The <dfn>means</dfn> are the general method of [=data processing=] through which a given
-[=purpose=] is implemented, in a given [=context=], considered at a relatively abstract
-level and not necessarily all the way down to implementation details. Example:
-<em>a person will have their preferences restored (purpose) by looking up their identifier
-in a preferences store (means)</em>.
+The <dfn>means</dfn> is the general way that data is [=processed=] to achieve a particular
+[=purpose=], in a given [=context=]. [=Means=] are relatively abstract
+and don't specify all the way down to implementation details. For example, for
+the [=purpose=] of restoring a person's preferences, the [=means=] could be to
+look up their identifier in a preferences store.
 
 A <dfn>data controller</dfn> is an [=actor=] that determines the [=means=] and [=purposes=]
 of data processing. Any [=actor=] that is not a [=service provider=] is a [=data controller=].
 
-A <dfn data-lt="data processor">service provider</dfn> or [=data processor=] is considered to be in
-the same category of [=first party=] or [=third party=] as the [=actor=] contracting it to
-perform the relevant [=processing=] if it:
+A <dfn data-lt="data processor">service provider</dfn> or [=data processor=]:
 
-* is processing the data on behalf of that [=actor=];
+* [=processes=] data on behalf of a [=data controller=];
```

Comment:
This is incorrect: you can have a service provider of a service provider, in fact it's very common.

Reply:
Fixed.